Rework of the Oath of Demon Hunter and add 2024 fighting style choose rule

magicskysword opened the pull request:
---

# Addition of 5e2024 Rule Options
This update adds optional 5e2024 rules allowing:
- Paladins and Rangers to select from all Fighter Combat Styles.

---

# New Tools
A new Python script has been added to assist with bulk file translation using an LLM-powered workflow, improving automation and consistency across multilingual content.

---

# Oath of Demon Hunter
Added new resources corresponding to the subclass features and abilities.

## Description

Paladins sworn to the Oath of Demon Hunter devote themselves to eliminating evil creatures lurking in the darkness and preserving peace in the mortal realm. Monsters are their quarry, whether spawned from the abyss, other planes, or ancient curses. They judge calmly, discerning “monster” from “human” before pulling the trigger. Where darkness gathers, they stand watch.

## Light Energy Crossbow Bolt (Level 3)
You gain the Light Energy Crossbow Bolt feature. You recover uses equal to your Charisma modifier per long rest and may expend one use of Channel Divinity to restore all uses. As a bonus action, expend one bolt to gain the following benefits for one minute:
- Use Charisma instead of Dexterity for attack and damage rolls with crossbows (light, heavy, hand).
- Ignore the Loading property when using crossbows.
- When attacking with crossbows, you may apply Divine Smite or other smite spells on hit.(compatible with 2014 and 2024)

## Channel Divinity: Trial Mark
When you hit an unmarked enemy with a crossbow attack, you may expend Channel Divinity to apply Trial Mark for 1 minute. Marked enemies cannot turn invisible. When attacking a marked target with a crossbow:
- Your critical threat range increases by 1.
- You deal additional radiant damage.

Additional radiant damage progression:
- 3rd: 1d4  
- 6th: 1d6  
- 10th: 1d8  
- 14th: 1d10  
- 18th: 1d12  

## Divine Crossbow (Level 7)
When using Light Energy Crossbow Bolt, gain the following:
- No disadvantage when attacking with crossbows while adjacent to hostile creatures.
- Your crossbow attacks may deal radiant damage instead of their normal damage type.

## Hunter’s Sight (Level 7)
You gain enhanced perception when fighting monsters:
- No disadvantage when attacking enemies in darkness.
- Add your Charisma modifier to Perception checks.

## Demon Hunter (Level 15)
You become a true hunter of abominations:
- When attacking a creature marked by Trial Mark, you immediately gain its primary statistics (HP, attributes, CR, AC, saving throws, resistances, immunities, vulnerabilities) and ignore its radiant resistance.
- Once per round, after hitting an enemy, you may use Hunter’s Step to teleport up to 30 ft without expending an action.

## Extra Attack (2) (Level 15)
When taking the Attack action, you can attack three times instead of once.

## Demonslayer (Level 20)
You embody the pinnacle of radiant destruction:
- When rolling radiant damage dice, reroll all dice showing less than 3; rerolled dice have a minimum value of 3.
- When a creature affected by Trial Mark dies, regain one use of Channel Divinity.

magicskysword committed:
Remaster Oath Of Demon Hunter

magicskysword committed:
Remaster Oath Of Demon Hunter lv 7

magicskysword committed:
Remaster Oath Of Demon Hunter lv 7-2

magicskysword committed:
Remaster Oath Of Demon Hunter lv 15

magicskysword committed:
Remaster Oath Of Demon Hunter lv 20

magicskysword committed:
Allow 2014 Oath Of Demon Hunter use crossbow cast smite spell

magicskysword committed:
Fix Oath Of Demon Hunter Bug and add feature image

magicskysword committed:
Add 2024 fighting style setting

magicskysword committed:
Add LLM script to help with translation

magicskysword committed:
Update Oath Of Demon Hunter localization file

magicskysword committed:
Merge branch 'dev' into oath_of_demon_hunter_remaster

magicskysword committed:
Merge branch 'dev' into oath_of_demon_hunter_remaster

magicskysword committed:
Add fighting style 2024 setting to GeneralDisplay

magicskysword committed:
Fixed Oath Of Demon Hunter some minor bug

EnderWiggin commented:
First of all - I don't like that there's 3 PRs worth in this one PR - please split them up.
Secondly - these are massive changes to the Demon Hunter - it will break all already created characters. Better to make it completely new subclass that is either alongside old one, or hide old one from selection, but present in the code -  that way in progress games can continue without interruptions.

EnderWiggin commented on SolastaUnfinishedBusiness/Behaviors/Specific/_GLBM.cs:
```diff
@@ -1014,6 +1014,17 @@ when attacker.UsedSpecialFeatures.ContainsKey(featureDefinition.Name):
                             {
                                 break;
                             }
+                            // Check Light Energy Crossbow Bolt condition is active for Demon Hunter
```
Same check is present in the `SmiteSpells2024Context.cs` - move it static method in subclass file and call from here.

EnderWiggin commented on SolastaUnfinishedBusiness/Models/SmiteSpells2024Context.cs:
```diff
@@ -125,19 +133,21 @@ internal static IEnumerator OnAttackHitConfirmed(GameLocationBattleManager battl
         {
-            //Demon Hunter can use Divine Smite on crossbows, but not other Smite spells
-            if (OathOfDemonHunter.IsOathOfDemonHunterWeapon(attackMode, null, attacker.RulesetCharacter))
+            //Demon Hunter can use Smite spells on crossbows
```
Same check is present in `__GLBM.cs` - move it to the static method in subclass file and call here.

EnderWiggin commented on SolastaUnfinishedBusiness/Patches/CharacterActionAttackPatcher.cs:
```diff
@@ -929,10 +930,16 @@ internal static IEnumerator ExecuteImpl(CharacterActionAttack __instance)
 
             //PATCH: supports smite spell scenarios
-            if (attackHasDamaged && !rangeAttack)
+            // Allow Oath of Demon Hunter to trigger smite concentration end with crossbows
```
Move demonhunter attack check to static method in subclass file and call here

magicskysword commented:
Thank you for the review. I’ve updated the relevant parts and split the work into three separate PRs:

- #76 **Add 2024 Fighting Style Rule**
- #75 **Add new LLM translate tool script**
- #74 **Remaster Oath of Demon Hunter**
